test/adapters/kyc-onboarding.test.js

```diff
@@ -418,11 +454,131 @@ describe("Adapter - KYC Onboarding", () => {
+    const { dao, adapters, extensions } = await deployDefaultDao({
+      owner: daoOwner,
+      kycSignerAddress: signer.address,
+      kycCanTopUp: 1,
+    });
+
+    const applicant = accounts[2];
+    const onboarding = adapters.kycOnboarding;
+    const bank = extensions.bankExt;
+    const memberNonce =
+      (await onboarding.memberNonces(applicant)).toNumber() + 1;
+    const ethAmount = unitPrice.mul(toBN(3)).add(remaining);
+    const signerUtil = SigUtilSigner(signer.privKey);
+
+    const couponData = {
+      type: "coupon-kyc",
+      kycedMember: applicant,
+      memberNonce,
+    };
+
+    const signature = signerUtil(
+      couponData,
+      dao.address,
+      onboarding.address,
+      chainId
+    );
+
+    await onboarding.onboardEth(
+      dao.address,
+      applicant,
+      memberNonce,
+      signature,
+      {
+        from: applicant,
+        value: ethAmount,
+        gasPrice: toBN("0"),
+      }
+    );
+
+    const applicantIsActiveMember = await isMember(bank, applicant);
+    expect(applicantIsActiveMember).equal(true);
+  });
+
+  it("should be possible for an existing member to top up when configured with canTopUp", async () => {
+    const { dao, adapters, extensions } = await deployDefaultDao({
+      owner: daoOwner,
+      kycSignerAddress: signer.address,
+      kycCanTopUp: 1,
+    });
+
+    const applicant = accounts[2];
+    const onboarding = adapters.kycOnboarding;
+    const bank = extensions.bankExt;
+    const memberNonce =
+      (await onboarding.memberNonces(applicant)).toNumber() + 1;
+    const ethAmount = unitPrice.mul(toBN(3)).add(remaining);
+    const signerUtil = SigUtilSigner(signer.privKey);
+
+    const couponData = {
+      type: "coupon-kyc",
+      kycedMember: applicant,
+      memberNonce,
+    };
+
+    const signature = signerUtil(
+      couponData,
+      dao.address,
+      onboarding.address,
+      chainId
+    );
+
+    await onboarding.onboardEth(
+      dao.address,
+      applicant,
+      memberNonce,
+      signature,
+      {
+        from: applicant,
+        value: ethAmount,
+        gasPrice: toBN("0"),
+      }
+    );
+
+    let applicantUnits = await bank.balanceOf(applicant, UNITS);
+    expect(applicantUnits.toString()).equal(
+      numberOfUnits.mul(toBN("3")).toString()
+    );
+
+    const couponData2 = {
+      type: "coupon-kyc",
+      kycedMember: applicant,
+      memberNonce: memberNonce + 1,
+    };
+
+    const signature2 = signerUtil(
+      couponData2,
+      dao.address,
+      onboarding.address,
+      chainId
+    );
+
+    await onboarding.onboardEth(
+      dao.address,
+      applicant,
+      memberNonce + 1,
+      signature2,
+      {
+        from: applicant,
+        value: ethAmount,
+        gasPrice: toBN("0"),
+      }
+    );
+
+    applicantUnits = await bank.balanceOf(applicant, UNITS);
+    expect(applicantUnits.toString()).equal(
+      numberOfUnits.mul(toBN("6")).toString()
+    );
+  });
```

Comment:
I didn't see that but we could add a test case to cover the coupon replay. WDYT?

Reply:
yes I'll add